sycl/test-e2e/Benchmarks/SYCLFrameworkOptimization/FloydWarshall.cpp

```diff
@@ -0,0 +1,365 @@
+// Test checks SYCL Framework optimization feature.
+// The program is being compiled with optimization and without.
+// Then the execution time of both are being relatively compared.
+//
+// The source code is taken from
+// https://github.com/oneapi-src/oneAPI-samples/blob/master/DirectProgramming/C%2B%2BSYCL/GraphAlgorithms/all-pairs-shortest-paths/src/apsp.cpp
```

Comment:
Don't we need a license header here then?

Reply:
Unfortunately, I am not a legal expert. The original code is owned by Intel according to the license header.

Reply:
@maksimsab you can always contact our legal folks to get an advice (shared the contacts internally).

From what I see, the file has MIT license. MIT license has the following statement:

```
The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.
```

So, MIT license requires to add copyright notice and license header to such file we are discussing which has portion of code from the file which is under MIT license.